model/Resource/Service/ItemRelationsService.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2025 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoAdvancedSearch\model\Resource\Service;
+
+use oat\tao\model\resources\relation\FindAllQuery;
+use oat\tao\model\resources\relation\ResourceRelation;
+use oat\tao\model\resources\relation\ResourceRelationCollection;
+use oat\tao\model\resources\relation\service\ResourceRelationServiceInterface;
+use oat\taoAdvancedSearch\model\SearchEngine\Service\ItemUsageService;
+
+class ItemRelationsService implements ResourceRelationServiceInterface
+{
+    private ItemUsageService $itemUsageService;
+
+    public function __construct(ItemUsageService $itemUsageService)
+    {
+        $this->itemUsageService = $itemUsageService;
+    }
+
+    public function findRelations(FindAllQuery $query): ResourceRelationCollection
+    {
+        $resourceRelationCollection = new ResourceRelationCollection();
```

Comment:
Lets call `search` method here directly here as it is the only necessary thing to do https://github.com/oat-sa/extension-tao-advanced-search/pull/140/files#diff-3af9fe84e20793c98cd35995836e8b435d33b81237207734ce7cdd019a62f363L95

The other services and `boolQuery` methods are not necessary